deployment/src/main/java/io/quarkus/jgit/deployment/JGitBuildTimeConfig.java

```diff
@@ -46,6 +47,12 @@ interface DevService {
         @WithDefault("quarkus")
         String adminPassword();
 
+        /**
+         * Repositories to be created when the Dev Service starts.
+         */
+        @WithDefault("${quarkus.application.name}")
+        List<String> repositories();
```

Comment:
You're right, it needs to be an optional when `quarkus.jgit.devservices.repositories=` is used in the `application.properties`

```suggestion
        Optional<List<String>> repositories();
```